app/src/forms/email/emailService.js

```diff
@@ -3,176 +3,110 @@ const fs = require('fs');
 const log = require('npmlog');
 const path = require('path');
 const constants = require('../common/constants');
-
 const formService = require('../form/service');
 
-const service = {
+// Helper function used to generate contexts for the ches.merge data argument
+const generateContexts = (type, configData, submission, referer) => {
+  let contextToVal = [];
+  if (type === 'sendStatusAssigned') {
+    contextToVal = [configData.assignmentNotificationEmail];
+  } else if (type === 'sendSubmissionConfirmation' && configData.form.showSubmissionConfirmation) {
+    contextToVal = [configData.body.to];
+  } else if (type === 'sendSubmissionReceived' && configData.form.submissionReceivedEmails) {
+    contextToVal = configData.form.submissionReceivedEmails;
+  }
+  const contexts = [
+    {
+      context: {
+        confirmationNumber: submission.confirmationId,
+        title: configData.title,
+        messageLinkText: configData.messageLinkText,
+        messageLinkUrl: `${service._appUrl(referer)}/form/view?s=${submission.id}`,
```

Comment:
We'll have to adjust this so the submission confirmation email sends the user back to /success?

Reply:
Good catch - I completely missed this.

Reply:
`


<!--StartFragment-->const generateContexts = (type, configData, submission, referer) =&gt; {  let contextToVal = [];  let userTypePath = '';  if (type === 'sendStatusAssigned') {    contextToVal = [configData.assignmentNotificationEmail];  } else if (type === 'sendSubmissionConfirmation' &amp;&amp; configData.form.showSubmissionConfirmation) {    contextToVal = [configData.body.to];    userTypePath = '/view?s=';  } else if (type === 'sendSubmissionReceived' &amp;&amp; configData.form.submissionReceivedEmails) {    contextToVal = configData.form.submissionReceivedEmails;    userTypePath = '/success?s=';  }  return [{    context: {      confirmationNumber: submission.confirmationId,      title: configData.title,      messageLinkText: configData.messageLinkText,      messageLinkUrl: `${service._appUrl(referer)}/form${userTypePath}${submission.id}`,    },    to: contextToVal  }];};<!--EndFragment-->

`

I added some logic to the generateContexts function in the emailService.js to append /view or /success to those different situtations.